src/Panel/Model.php

```diff
@@ -195,22 +212,39 @@ protected function imageSource(?string $query = null)
 
         // validate the query result
         if (
-            is_a($image, 'Kirby\Cms\File') === false &&
-            is_a($image, 'Kirby\Filesystem\Asset') === false
+            is_a($image, 'Kirby\Cms\File') === true ||
+            is_a($image, 'Kirby\Filesystem\Asset') === true
         ) {
-            $image = null;
+            return $image;
         }
 
-        // fallback for files
-        if (
-            $image === null &&
-            is_a($this->model, 'Kirby\Cms\File') === true &&
-            $this->model->isViewable() === true
```

Comment:
What happened to this viewable check? Has it been moved elsewhere or is it no longer needed?

Reply:
It's still here https://github.com/getkirby/kirby/blob/feature/inertia/src/Panel/File.php#L155 - more specific to the `File` object. The previous implementation was a bit redundant.